test/test_statsd_tcp.js

```diff
@@ -5,666 +5,628 @@ var net = require('net');
 var StatsD = require('../').StatsD;
 
 /**
- * Creates a test harness for TCP, that binds to an ephemeral port
- * @param test {Function} The test to run, should take message as the argument
- * @param callback {Function} The callback to call after the server is listening
- * @private
+ * Since sampling uses random, we need to patch Math.random() to always give
+ * a consistent result
  */
-function tcpTest(test, callback) {
+Math.random = function () {
+  return 0.42;
+};
 
+function createServer(onListening) {
   // This is the actual TCP server implementation of etsy/statsd
   // Including the metrics handler which splits the TCP packet on "\n"
   // to deal with multiple metrics in a single TCP stream data flush
-  var server = net.createServer(function (stream) {
-    stream.setEncoding('ascii');
+  var server = net.createServer(function (socket) {
+    socket.setEncoding('ascii');
 
     var buffer = '';
-    stream.on('data', function (data) {
+    socket.on('data', function (data) {
       buffer += data;
       var offset = buffer.lastIndexOf("\n");
       if (offset > -1) {
         var packet = buffer.slice(0, offset + 1);
         buffer = buffer.slice(offset + 1);
 
-        var metrics;
+        var parts;
         if (packet.indexOf("\n") > -1) {
-          metrics = packet.split("\n");
+          parts = packet.split("\n");
         } else {
-          metrics = [packet];
+          parts = [packet];
         }
 
-        metrics.forEach(function (metric) {
-          test(metric, server);
-        });
+        var metrics = parts.filter(part => part !== '');
```

Comment:
test/test_statsd_tcp.js: line 37, col 41, 'arrow function syntax (=>)' is only available in ES6 (use 'esversion: 6').